extensions/windows-event-log/ConsumeWindowsEventLog.h

```diff
@@ -101,14 +100,11 @@ class ConsumeWindowsEventLog : public core::Processor
   
 
 protected:
-  bool subscribe(const std::shared_ptr<core::ProcessContext> &context);
-  void unsubscribe();
-  int processQueue(const std::shared_ptr<core::ProcessSession> &session);
+  void putEventRenderFlowFileToSession(const EventRender& eventRender, core::ProcessSession& session);
   wel::WindowsEventLogHandler getEventLogHandler(const std::string & name);
   bool insertHeaderName(wel::METADATA_NAMES &header, const std::string &key, const std::string &value);
   void LogWindowsError();
-  void processEvent(EVT_HANDLE eventHandle);
-  bool processEventsAfterBookmark(EVT_HANDLE hEventResults, const std::wstring& channel, const std::wstring& query);
+  bool createEventRender(EVT_HANDLE eventHandle, EventRender& eventRender);
```

Comment:
A small suggestion: "Throw an exception to signal that a function can’t perform its assigned task" (instead of returning success/failure `bool`)
This way `eventRender` can become the return value, making the function more intuitive to use.

http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#e2-throw-an-exception-to-signal-that-a-function-cant-perform-its-assigned-task

Reply:
Agree, but since it is minor will leave it as it is.